lib/timeline/component/item/RangeItem.js

```diff
@@ -109,27 +104,91 @@ RangeItem.prototype.redraw = function() {
     var className = (this.data.className ? (' ' + this.data.className) : '') +
         (this.selected ? ' vis-selected' : '') + 
         (editable ? ' vis-editable' : ' vis-readonly');
-    dom.box.className = this.baseClassName + className;
-
-    // determine from css whether this box has overflow
-    this.overflow = window.getComputedStyle(dom.frame).overflow !== 'hidden';
+    this.dom.box.className = this.baseClassName + className;
 
-    // recalculate size
     // turn off max-width to be able to calculate the real width
     // this causes an extra browser repaint/reflow, but so be it
     this.dom.content.style.maxWidth = 'none';
-    this.props.content.width = this.dom.content.offsetWidth;
-    this.height = this.dom.box.offsetHeight;
-    this.dom.content.style.maxWidth = '';
+  }
+}
 
-    this.dirty = false;
+RangeItem.prototype._getDomComponentsSizes = function() {
+  // determine from css whether this box has overflow
+  this.overflow = window.getComputedStyle(this.dom.frame).overflow !== 'hidden';
+  return {
+    content: {
+      width: this.dom.content.offsetWidth,
+    },
+    box: {
+      height: this.dom.box.offsetHeight
+    }
   }
+}
+
+RangeItem.prototype._updateDomComponentsSizes = function(sizes) {
+  this.props.content.width = sizes.content.width;
+  this.height = sizes.box.height;
+  this.dom.content.style.maxWidth = '';
+  this.dirty = false;
+}
 
-  this._repaintOnItemUpdateTimeTooltip(dom.box);
-  this._repaintDeleteButton(dom.box);
+RangeItem.prototype._repaintDomAdditionals = function() {
+  this._repaintOnItemUpdateTimeTooltip(this.dom.box);
+  this._repaintDeleteButton(this.dom.box);
   this._repaintDragCenter();
   this._repaintDragLeft();
   this._repaintDragRight();
+}
+
+
+/**
+ * Repaint the item
+ * @param {boolean} [returnQueue=false]  return the queue
+ * @return {boolean} the redraw queue if returnQueue=true
+ */
+RangeItem.prototype.redraw = function(returnQueue) {
```

Comment:
Strike four: Image Samuel L. Jackson holding a gun against your head and what he would say about DRY-ing this.